File: Java/util/InputOption.java
package util;

import java.util.Scanner;

public class InputOption extends InputValidator{
	private static Scanner scan = new Scanner(System.in);
	
	public int inputProgramContinue(){
		String continueProgram = "";
		do {
			System.out.printf("\nYour answer: ");
			continueProgram = scan.nextLine();
			if (!(continueProgram.equalsIgnoreCase("Y") || continueProgram.equalsIgnoreCase("N"))){
				System.out.printf("Please input the proper answer!\n");
			}
			else {
				if (continueProgram.equalsIgnoreCase("N")){
					return 99;
				}
			}
		}
		while (!(continueProgram.equalsIgnoreCase("Y") || continueProgram.equalsIgnoreCase("N")));
		return 1;
	}
	
	public int inputContinueFlow(int choiceLimit){
		int choice;
		System.out.printf("\nYour number of choice: ");
		do {
			choice = safeInteger();
			if (choice != 99 && (choice < 0 || choice > choiceLimit)){
				System.out.printf("Input the proper number of choice: ");
			}
		}
		while (choice != 99 && (choice < 0 || choice > choiceLimit));
		return inputBackOrStop(choice);
	}
	
	private int inputBackOrStop(int choice){
		switch (choice){
			case 0:
				return -1;
			case 99:
				return 99;
		}
		return choice;
	}
}

File: Go/src/util/input/inputOption.go
package input

import (
	"fmt"
	"strings"
)

func InputProgramContinue() int {
	continueProgram := ""
	for isOk := true; isOk; isOk = !(strings.EqualFold(continueProgram, "Y") || strings.EqualFold(continueProgram, "N")) {
		fmt.Printf("\nYour answer: ")
		continueProgram = InputValue()
		if !(strings.EqualFold(continueProgram, "Y") || strings.EqualFold(continueProgram, "N")) {
			fmt.Printf("Please input the proper answer!\n")
			continue
		}
	}
	if strings.EqualFold(continueProgram, "N") {
		return 99
	}
	return 1
}

func InputContinueFlow(choiceLimit int) int {
	var choice int
	fmt.Printf("\nYour number of choice: ")
	for isOk := true; isOk; isOk = (choice != 99 && (choice < 0 || choice > choiceLimit)) {
		choice = SafeInteger()
		if choice != 99 && (choice < 0 || choice > choiceLimit) {
			fmt.Printf("Input the proper number of choice: ")
		}
	}
	return inputBackOrStop(choice)
}

func inputBackOrStop(choice int) int {
	switch choice {
	case 0:
		return -1
	case 99:
		return 99
	}
	return choice
}
